guppy/src/graph/build.rs

```diff
@@ -1002,6 +1008,42 @@ fn convert_forward_slashes<'a>(rel_path: impl Into<Cow<'a, Utf8Path>>) -> Utf8Pa
     }
 }
 
+// Calculate the relative path from `from` to `to`.
+// This function finds the relative path between two given paths.
+// It first identifies the common prefix between the two paths and then
+// constructs the relative path by adding ".." for each remaining component
+// in the `from` path and appending the remaining components from the `to` path.
+#[track_caller]
+pub fn find_relative_path_utf8(from: &Utf8Path, to: &Utf8Path) -> Utf8PathBuf {
```

Comment:
This function should live on `PackageSourceImpl`, next to `PackageSourceImpl::create_path` which does something similar for path dependencies.